subsys/bluetooth/controller/ll_sw/ull_llcp_conn_upd.c

```diff
@@ -326,6 +389,13 @@ static void lp_cu_st_wait_rx_conn_param_rsp(struct ll_conn *conn, struct proc_ct
 
 	switch (evt) {
 	case LP_CU_EVT_CONN_PARAM_RSP:
+		llcp_pdu_decode_conn_param_rsp(ctx, param);
+		/* Perform Param check and possibly reject (LL_REJECT_EXT_IND) */
+		if (!cu_check_conn_parameters(conn, ctx)) {
+			ctx->data.cu.error = BT_HCI_ERR_INVALID_LL_PARAM;
+			lp_cu_send_reject_ext_ind(conn, ctx, evt, param);
+			break;
+		}
```

Comment:
You still need call `llcp_rr_set_incompat(conn, INCOMPAT_RESERVED);` as you have received the first PDU. Also you need to call `llcp_rr_set_incompat(conn, INCOMPAT_NO_COLLISION);` when the procedure is done, even if rejecting; otherwise the collision detection system will be lost.

Reply:
Thanks! I missed that.